build_debian.sh

```diff
@@ -392,11 +392,17 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
     systemd-sysv \
     ntp
 
-if [[ $CONFIGURED_ARCH == amd64 ]]; then
+if [[ $TARGET_BOOTLOADER == grub ]]; then
+    if [[ $CONFIGURED_ARCH == amd64 ]]; then
+        GRUB_PKG=grub-pc-bin
+    elif [[ $CONFIGURED_ARCH == arm64 ]]; then
+        GRUB_PKG=grub-efi-arm64-bin
+    fi
```

Comment:
armhf is irrelevant here because these lines are concerning the grub package to install from APT. 

There are two types of platforms which now use grub in SONiC:
1. All amd64 platforms
2. Mellanox arm64 platforms

No armhf platforms support grub. As that would be packages and code I would be unable to test I do not want to include it, that instead would be the responsibility of someone trying to add support for a platform which supports grub on armhf.